scripts/spyder_win_post_install.py

```diff
@@ -21,17 +21,95 @@
 KEY_C1 = KEY_C0 + r"\command"
 
 
+# ability to run spyder-win-post-install outside of bdist_wininst installer
+# copied from pywin32-win-post-install
+try:
+    # When this script is run from inside the bdist_wininst installer,
+    # file_created() and directory_created() are additional builtin
+    # functions which write lines to Python23\pywin32-install.log. This is
+    # a list of actions for the uninstaller, the format is inspired by what
+    # the Wise installer also creates.
+    file_created
+    is_bdist_wininst = True
+except NameError:
+    is_bdist_wininst = False # we know what it is not - but not what it is :)
+    def file_created(file):
+        pass
+    def directory_created(directory):
+        pass

```

Comment:
When the user runs the `spyder_win_post_install.py` manually after using pip to install Spyder, there isn't a registered uninstaller for Spyder. The uninstaller is only registered when using the `bdist_wininst` installer. The functions [`file_created()` and `directory_created()`](https://docs.python.org/2/distutils/builtdist.html#the-postinstallation-script) normally register files and directories to be removed with the uninstaller, but for pip users these functions should do nothing. Instead if the shortcut files and startmenu directory exist, they are removed in [L174](https://github.com/spyder-ide/spyder/pull/2377/files#diff-044873a159653f78071aafd4f514e937R174) and [L186](https://github.com/spyder-ide/spyder/pull/2377/files#diff-044873a159653f78071aafd4f514e937R186).